contracts/monitoring/src/contract.rs

```diff
@@ -37,110 +38,216 @@ pub fn execute(
             execute::check_governance(&deps, info)?;
             execute::register_prover(deps, chain_name, new_prover_addr)
         }
+        ExecuteMsg::RegisterActiveWorkerSet { next_worker_set } => {
+            // TODO: add check_prover to make sure prover is part of the system
```

Comment:
I actually think it's not totally necessary to check that. Internally you are just mapping this worker set to the sender address. If some random address sends in a worker set, that's not really going to cause a problem, because the contract is just going to associate a worker set with that address, but the address will never be associated with a chain.